torch/_inductor/codegen/wrapper.py

```diff
@@ -1702,9 +1741,10 @@ def codegen_model_kernels(self):
             "class AOTInductorModelKernels : public AOTInductorModelKernelsBase {"
         )
         self.prefix.writeline("  public:")
-        for kernel in chain(
-            self.src_to_kernel.values(), self.user_defined_kernel_cache.values()
-        ):
+        declare_kernel = set(self.src_to_kernel.values())
```

Comment:
I don't think it matters w.r.t. `chain`, since the values are just the keys' string (which would be plugged in `set`). I use set here so that we won't "redeclare" kernels that are shared by both constant's code and main code.